src/main/java/com/android/volley/toolbox/HurlStack.java

```diff
@@ -254,7 +254,7 @@ private static void addBody(HttpURLConnection connection, Request<?> request, by
         // since this is handled by HttpURLConnection using the size of the prepared
         // output stream.
         connection.setDoOutput(true);
-        connection.addRequestProperty(
+        connection.setRequestProperty(
```

Comment:
Hi @houtianze @jpd236,
There have been a lot of issues / confusion related to devs setting the Content-Type header using the getHeaders method instead of overriding the getBodyContentType (maybe also due to lack of docs at https://developer.android.com/training/volley/request-custom.html). Doing that would result in Volley sending request with two Content-Type headers. Usually, servers don't strictly verify this, but in a current ongoing issue we're having that had been very hard to zero in on, MS Azure servers/gateways decline this with a 400 BAD request.

This change should fix the issue, but the users may be confused as to why the Content-Type header is not being set to what they explicitly passed, if they are overriding getHeaders and not getBodyContentType. Would it make sense to do a precheck here for Content-Type header and inform the users about the same (log messages?) and then go ahead with replacing the header?

Reply:
Ah never mind, just saw the conversation. Moving this comment there.